sotodlib/io/load_smurf.py

```diff
@@ -1683,6 +1703,42 @@ def get_channel_info(status, mask=None, archive=None, obsfiledb=None,
     
     return ch_info
 
+def _get_sample_info(filenames):
+    """Scan through a list of files and count samples.
+    
+    Args
+    -----
+    filenames : list
+        list of filenames
+    
+    Returns 
+    --------
+    out : list
+        a list of dictionaries formatted for load_file. 
+        in the pattern of [ {filename: filename, sample_range: (sample_start, sample_stop)}]
```

Comment:
This was definitely slower when I ran it on a bigger datafile but also I agree that I think that doesn't matter, since the usage is far more niche, especially since it's only likely to be done on one file at a time (or O(1), at least). I think the implementation seems to work, so I'm fine with this, provided the docstring is updated which I think you already did.

Reply:
Yeah, I think the much more common usages will be either (1) no database but not sending in sample cuts or (2) sample cuts + database. So it's probably fine? :-D